apps/meteor/client/views/marketplace/hooks/useAppMenu.tsx

```diff
@@ -118,10 +122,28 @@ export const useAppMenu = (app: App, isAppDetailsPage: boolean) => {
 		setIsPurchased: setPurchased,
 	});
 
+	const addonHandler = useCallback(
```

Comment:
Since most of the `useCallback`s here are to be used on an event, maybe we could swap them out for `useEffectEvent`?

Reply:
this callbacks will not be used by any effect... they will be part of the onClick event. They are a cascade of callbacks...

Reply:
There is a obscene amount of dependencies for a single `useMemo` at the end. @MartinSchoeler's approach would benefit a little, but I'd suggest, for now, to move the callbacks into that `useMemo`. The long the solution would be to break this huge hook into smaller ones.

Reply:
Updating with a TODO comment for a future refactor.

Reply:
@tiagoevanp @MartinSchoeler  just to ensure you got it right... Effects are reactive pieces of code that let you use non-React logic inside a React component, while event handlers are triggered by any kind of event that is not a React state mutation. We should avoid effects that depend on event handlers to be triggered e.g. passing a new value of `onClick` should not trigger a new `fetch` call from the inner workings of a component. But, because `useEffectEvent` callbacks are stable, we tend to abuse this hook as a memoization mechanism (in this case, we're trying to avoid a `useMemo` invalidation). The alternative for not doing this kind of premature optimization is to passing primitive props (numbers, strings, booleans) to components instead of structured values that are passed by reference (objects, arrays, functions). In this scenario React is going to be smart enough to bail out of unneeded re-renders or the components receiving the props should be able to decide when to bail out (e.g. the component itself will envelope the `onClick` prop to use it inside an effect).